tools/swift-inspect/Tests/swiftInspectTests/swiftInspectTests.swift

```diff
@@ -12,48 +12,93 @@
 
 import XCTest
 import class Foundation.Bundle
+import libswiftInspect
 
 final class swiftInspectTests: XCTestCase {
-    func testExample() throws {
-        // This is an example of a functional test case.
-        // Use XCTAssert and related functions to verify your tests produce the correct
-        // results.
+  func testNoConcurrencyActivity() throws {
+    let output = try dumpConcurrency()
+    let (tasks, actors, threads) = try split(output: output)
+    XCTAssertEqual(tasks, [""])
+    // FIXME: sometimes we get a bunch of objects misinterpreted as actors in the actor output
+    XCTAssertEqual(threads, ["  no threads with active tasks"])
+  }
 
-        // Some of the APIs that we use below are available in macOS 10.13 and above.
-        guard #available(macOS 10.13, *) else {
-            return
-        }
+  func testBasicOutput() async throws {
+    actor BasicOutputTestActor {}
+    let actor = BasicOutputTestActor()
+    defer { withExtendedLifetime(actor) {} }
 
-        let fooBinary = productsDirectory.appendingPathComponent("swift-inspect")
+    let output = try dumpConcurrency()
+    let (tasks, actors, threads) = try split(output: output)
 
-        let process = Process()
-        process.executableURL = fooBinary
+    XCTAssertGreaterThan(count(ofSubstring: "Task 0x", in: tasks), 0,
+                         "No tasks present in task output: \(tasks.joined(separator: "\n"))")
+    XCTAssertGreaterThan(count(ofSubstring: "BasicOutputTestActor", in: actors), 0,
+                         "No actors present in actor output: \(actors.joined(separator: "\n"))")
+    XCTAssertGreaterThan(count(ofSubstring: "Thread 0x", in: threads), 0,
+                         "No threads present in thread output: \(threads.joined(separator: "\n"))")
+  }
 
-        let pipe = Pipe()
-        process.standardOutput = pipe
+  private func dumpConcurrency() throws -> [String] {
```

Comment:
Please guard this with an os check.  The `--fork-corpse` option is not global.

Reply:
Right you are. I'm guessing `getpid` is not going to fly on Windows either, what should I use there?

Reply:
Right, `getpid` doesn't work.  You could use `GetProcessId(GetCurrentProcess())` to get the equivalent for that.

https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getcurrentprocess
https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getprocessid

Reply:
Fixed.